src/main/java/org/broadinstitute/hellbender/tools/spark/SparkReadShard.java

```diff
@@ -0,0 +1,52 @@
+package org.broadinstitute.hellbender.tools.spark;
+
+
+import org.broadinstitute.hellbender.engine.ReadShard;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.Serializable;
+import java.util.Iterator;
+
+public class SparkReadShard implements ReadShard, Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final ShardBoundary key;
```

Comment:
Oh I see, it's an inner class of ReadShard. That's too fancy. Can you take that class out and make it a top-level? It does not seem to need to be an non-static inner class (ie does not the pointer to enclosing instance as far as i can tell)